core/modules/user/user.module

```diff
@@ -1044,6 +1060,117 @@ function user_account_form_validate($form, &$form_state) {
   }
 }
 
+/**
+ * Validation callback for password constraints.
+ *
+ * @see user_account_form()
+ * @see user_pass_reset_form()
+ */
+function user_password_constraints_validate($form, &$form_state) {
+  $strength = 0;
+  if (!array_key_exists('pass', $form_state['values'])) {
+    // If mail verification is required, user registration forms don't have a
+    // password field.
+    return;
+  }
+  $password = trim($form_state['values']['pass']);
+
+  if ($form['#form_id'] == 'user_profile_form' && strlen($password) == 0) {
+    // On profile form and no new password has been set.
+    return;
+  }
+  $config = config('system.core');
+  $messages = array();
+  if ($form['#form_id'] == 'user_pass_reset_form') {
+    $account = $form['#account'];
+    $username = $account->name;
+    $email = $account->mail;
+  }
+  else {
+    $username = $form_state['values']['name'];
+    $email = $form_state['values']['mail'];
+  }
+
+  $strength = _user_estimate_password_entropy($password);
+  if ($strength <= 50) {
+    $messages[] = t('The password is too weak. Please consider making your password more complex.');
+  }
+
+  $min_length_constraint = $config->get('user_password_min_length_constraint');
+  $min_length = $config->get('user_password_min_length');
+  // We need to match the JS string length method which uses UTF-16.
+  if ($min_length_constraint && (mb_strlen($password, 'UTF-16') < $min_length)) {
+    $strength = 5;
+    $messages[] = t('The password is too short. At least @min_length_length characters required.', array('@min_length' => $min_length));
+  }
+  if (backdrop_strtolower($password) == backdrop_strtolower($username)) {
+    $strength = 5;
+    $messages[] = t('The password cannot be the same as the username');
+  }
+  if (backdrop_strtolower($password) == backdrop_strtolower($email)) {
+    $strength = 5;
+    $messages[] = t('The password cannot be the same as the email');
+  }
+
+  if ($strength <= 50) {
+    $form_error = implode('<br>', $messages);
```

Comment:
Should we use `theme('item_list', ...)` here instead?

Reply:
I took a different approach of having each line set an error. The problem with using theme_item_list here is that it ends up with two bullets in a row.